### PR TITLE
Minor chain simulator refactor & unit tests

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Reasoning behind the pull request
- minor chain simulator refactor
- added unit tests
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: node/chainSimulator/chainSimulator.go [hunk 1]
```diff
@@ -412,30 +421,119 @@ func (s *simulator) SetStateMultiple(stateSlice []*dtos.AddressState) error {
...
+		txsAreExecuted := s.computeTransactionStatus(transactionStatus)
+		if txsAreExecuted {
+			return getApiTransactionsFromResult(transactionStatus), nil
+		}
+	}
+
+	return nil, errors.New("something went wrong. Transaction(s) is/are still in pending")
+}
+
+func (s *simulator) computeTransactionStatus(status []*transactionWithResult) bool {
```
Comment (sstanculeanu) [inline node/chainSimulator/chainSimulator.go:477]:
```suggestion
func (s *simulator) computeTransactionsStatus(status []*transactionWithResult) bool {
```

File: node/chainSimulator/chainSimulator.go [hunk 2]
```diff
@@ -412,30 +421,119 @@ func (s *simulator) SetStateMultiple(stateSlice []*dtos.AddressState) error {
...
+	}
+
+	txHashHex := hex.EncodeToString(txHash)
+	_, err = node.GetFacadeHandler().SendBulkTransactions([]*transaction.Transaction{tx})
+	if err != nil {
+		return "", err
+	}
+
+	for {
+		txs, _ := node.GetFacadeHandler().GetTransactionsPool("")
```
Comment (sstanculeanu) [inline node/chainSimulator/chainSimulator.go:528]: what if the tx is out of the pool before this line? is this possible and should we add an exit case here?

Comment (iulianpascalau) [reply node/chainSimulator/chainSimulator.go:528]: let's see, that was the code moved by me, indeed, but not the author. Will come back with an answer and solution to this.

Comment (iulianpascalau) [reply node/chainSimulator/chainSimulator.go:528]: clarified and made the code more robust

File: node/chainSimulator/chainSimulator.go (hunk 1 above)
Comment (iulianpascalau) [reply node/chainSimulator/chainSimulator.go:477]: fixed

**Review by sstanculeanu — DISMISSED** [review]

File: node/chainSimulator/chainSimulator.go [hunk 3]
```diff
@@ -412,30 +421,118 @@ func (s *simulator) SetStateMultiple(stateSlice []*dtos.AddressState) error {
...
+		txsAreExecuted := s.computeTransactionsStatus(transactionStatus)
+		if txsAreExecuted {
+			return getApiTransactionsFromResult(transactionStatus), nil
+		}
+	}
+
+	return nil, errors.New("something went wrong. Transaction(s) is/are still in pending")
+}
+
+func (s *simulator) computeTransactionsStatus(status []*transactionWithResult) bool {
```
Comment (miiu96) [inline node/chainSimulator/chainSimulator.go:477]: can you rename `status` in `txsWithResult` or something similar

Comment (iulianpascalau) [reply node/chainSimulator/chainSimulator.go:477]: renamed

**Review by sstanculeanu — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Pull request merged**